src/jockey/geometry.rs

```diff
@@ -0,0 +1,252 @@
+use std::collections::HashMap;
+
+use gl::types::*;
+
+/// A struct represents a single attribute of a Geometry.
+/// You should not use a single GeometryAttribute across multiple Geometries,
+/// since Geometry will delete the gl buffer on drop.
+pub struct GeometryAttribute<T> where T: std::fmt::Debug {
+    /// The data of the attribute.
+    pub array: Vec<T>,
+
+    /// The item size of the attribute.
+    pub size: GLuint,
+
+    /// The type of the attribute.
+    pub type_: GLenum,
+
+    /// Whether the data is normalized or not.
+    pub normalized: GLboolean,
+
+    /// Either `gl::ARRAY_BUFFER` or `gl::ELEMENT_ARRAY_BUFFER`
+    pub target: GLenum,
+
+    /// Usage of the attribute.
+    pub usage: GLenum,
+
+    /// A buffer object for this attribute.
+    _buffer: Option<GLuint>,
```

Comment:
In Rust, starting an identifier with an underscore signals the compiler that you don't care about this variable. As a consequence, rustc does not emit any warnings if the variable is unused. Since it looks like you need this variable, you should remove the underscore.

Reply:
I was too bound to JS/TS context 💦 Fixing